pkg/gardenlet/controller/seed/seed/reconciler.go

```diff
@@ -30,192 +41,110 @@ import (
 	gutil "github.com/gardener/gardener/pkg/utils/gardener"
 	"github.com/gardener/gardener/pkg/utils/imagevector"
 	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
-
-	"github.com/go-logr/logr"
-	corev1 "k8s.io/api/core/v1"
-	schedulingv1 "k8s.io/api/scheduling/v1"
-	apiequality "k8s.io/apimachinery/pkg/api/equality"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/client-go/tools/cache"
-	"k8s.io/client-go/tools/record"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-const reconcilerName = "seed"
-
-func (c *Controller) seedAdd(obj interface{}) {
-	key, err := cache.MetaNamespaceKeyFunc(obj)
-	if err != nil {
-		c.log.Error(err, "Could not get key", "obj", obj)
-		return
-	}
-	c.seedQueue.Add(key)
-}
-
-func (c *Controller) seedUpdate(oldObj, newObj interface{}) {
-	var (
-		oldSeed       = oldObj.(*gardencorev1beta1.Seed)
-		newSeed       = newObj.(*gardencorev1beta1.Seed)
-		specChanged   = !apiequality.Semantic.DeepEqual(oldSeed.Spec, newSeed.Spec)
-		statusChanged = !apiequality.Semantic.DeepEqual(oldSeed.Status, newSeed.Status)
-	)
-
-	if !specChanged && statusChanged {
-		return
-	}
-	c.seedAdd(newObj)
+// Reconciler reconciles Seed resources and provisions or de-provisions the seed system components.
+type Reconciler struct {
+	GardenClient                         client.Client
+	SeedClientSet                        kubernetes.Interface
+	Config                               config.GardenletConfiguration
+	Recorder                             record.EventRecorder
+	Identity                             *gardencorev1beta1.Gardener
+	ImageVector                          imagevector.ImageVector
+	ComponentImageVectors                imagevector.ComponentImageVectors
+	ClientCertificateExpirationTimestamp *metav1.Time
+	GardenNamespaceName                  string
```

Comment:
You have named the constant in lease reconciler struct as, LeaseNamespace and here it's GardenNamespaceName. Can we change this to GardenNamespace as well, for consistency?